dkg/disk.go

```diff
@@ -127,58 +125,26 @@ func writeDepositData(aggSigs map[core.PubKey]*bls_sig.Signature, withdrawalAddr
 }
 
 // checkWrites writes sample files to check disk writes and removes sample files after verification.
-func checkWrites(dataDir string, def cluster.Definition) error {
-	var shares []share
-	sigs := make(map[core.PubKey]*bls_sig.Signature)
-	for i := 0; i < def.NumValidators; i++ {
-		tss, sks, err := tbls.GenerateTSS(def.Threshold, len(def.Operators), rand.Reader)
-		if err != nil {
-			return err
+func checkWrites(dataDir string) error {
+	checkBody := []byte("delete me: dummy file used to check write permissions")
+	files := []string{"cluster-lock.json", "deposit-data.json", "validator_keys/keystore-0.json"}
+	for _, file := range files {
+		if strings.Contains(file, "/") {
+			if err := os.MkdirAll(path.Join(dataDir, path.Dir(file)), 0o777); err != nil {
+				return errors.Wrap(err, "mkdir check writes")
+			}
 		}
-
-		shares = append(shares, share{
-			PubKey:       tss.PublicKey(),
-			SecretShare:  sks[0],
-			PublicShares: tss.PublicShares(),
-		})
-
-		pk, err := tblsconv.KeyToCore(tss.PublicKey())
-		if err != nil {
-			return err
+		if err := os.WriteFile(path.Join(dataDir, file), checkBody, 0o444); err != nil {
+			return errors.Wrap(err, "write file check writes")
 		}
-
-		sig, err := testutil.RandomBLSSignature()
-		if err != nil {
-			return err
+		if err := os.Remove(path.Join(dataDir, file)); err != nil {
+			return errors.Wrap(err, "remove file")
+		}
+		if strings.Contains(file, "/") {
+			if err := os.RemoveAll(path.Join(dataDir, path.Dir(file))); err != nil {
```

Comment:
note that this directory might have existed before with other files, which you ar now deleting... probably not a good idea

Reply:
I would just skip deleting, since we are going to create the folder in anycase after this. Just add a `TODO: should we should delete the validator keys folder if we created it above?`

Reply:
done